### ISSUE TITLE
Support for user avatars

### ISSUE THREAD

**Issue by alphapapa** [alphapapa/ement.el#166]
Support for user avatars is available in a WIP branch here: https://github.com/alphapapa/ement.el/tree/wip/user-avatars  It seems to work pretty well, but it should be tested more carefully before being merged.

**Comment by Stebalien** [comment]
It would be nice if avatars were displayed in the margin when using elemental style sender headers. That way, the sender automatically aligns with the rest of the text.

It would be even better if we could have large multi-line avatars... but, this is emacs and doing _that_ would require deep magic and image slicing.

**Comment by alphapapa** [comment]
> It would be nice if avatars were displayed in the margin when using elemental style sender headers. That way, the sender automatically aligns with the rest of the text.

That would be nice, and it would imitate Element further, but it doesn't seem like Emacs handles images in margins very well.  But maybe this could be improved; patches welcome.

> It would be even better if we could have large multi-line avatars... but, this is emacs and doing _that_ would require deep magic and image slicing.

Yeah...that would be nice...but...

**Comment by Stebalien** [comment]
> That would be nice, and it would imitate Element further, but it doesn't seem like Emacs handles images in margins very well. But maybe this could be improved; patches welcome.

It worked for me in a quick hack, but getting it right-aligned seems more difficult.

(I probably won't get around to this any time soon, unfortunately).

**Comment by alphapapa** [comment]
> It worked for me in a quick hack, but getting it right-aligned seems more difficult.

Yeah, I've tried various combinations of display properties, but I also wasn't able to get them aligned in the margins.  (IIRC there's a comment somewhere in the Elisp manual that says that's not supported for images, but the details escape me.)  We might be able to approximate that by using the non-margin style with wrap-prefix properties.

**Comment by alphapapa** [comment]
> If this gets implemented, it'd be nice to have an option to disable the avatars.

Naturally.

**Comment by Thaodan** [comment]
What happened to this? Looking at the current option it almost looks like this already implemented but it isn't.
